app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java

```diff
@@ -1,24 +1,198 @@
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(requireActivity(), android.R.layout.simple_dropdown_item_1line, propertyType);
+        AutoCompleteTextView propertyTypeTextView = (AutoCompleteTextView) mBinding.addPropertyInformationLayout.addPropertyInformationType.getEditText();
+        assert propertyTypeTextView != null;
+        propertyTypeTextView.setAdapter(adapter);
+    }
+
+    //___________________________________CLICK LISTENER_____________________________________________
+
+    private void initClickListener() {
+        mBinding.addPropertyMediaLayout.detailCustomViewDeleteButton.setOnClickListener(v ->
+                Toast.makeText(requireContext(), "delete", Toast.LENGTH_SHORT).show());
+
+        mBinding.addPropertySoldLayout.addPropertySoldCheckbox.setOnClickListener(v ->
+                mBinding.addPropertySoldLayout.addPropertySoldEditText.setVisibility(
+                        mBinding.addPropertySoldLayout.addPropertySoldCheckbox.isChecked() ?
+                                View.VISIBLE : View.GONE));
+
+        mUpdateButton.setOnClickListener(mIsPropertyValuesAreNull ?
+                (v -> mPropertyViewModel.addProperty(updatedProperty(PROPERTY_TO_ADD(Objects.requireNonNull(mAgentViewModel.getAgent.getValue())))))
+                : (v -> mPropertyViewModel.updateProperty(updatedProperty(Objects.requireNonNull(mPropertyViewModel.getSelectedProperty.getValue())))));
+
+    }
+
+    public void configureOnClickRecyclerView() {
+        initAddMediaClickListener();
+        initSelectPhotoToDeleteOnLongClickListener();
+    }
+
+    private void initSelectPhotoToDeleteOnLongClickListener() {
+        RecyclerViewItemClickListener.addTo(mBinding.addPropertyMediaLayout.detailCustomViewRecyclerView, R.layout.fragment_add_property)
+                .setOnItemLongClickListener((recyclerView, position, v) -> {
+                    if (position == 0) return false;
+                    CheckBox selectToDeleteCheckBox = v.findViewById(R.id.property_detail_item_check_box);
+                    selectToDeleteCheckBox.setChecked(!selectToDeleteCheckBox.isChecked());
+                    selectToDeleteCheckBox.setVisibility(selectToDeleteCheckBox.isChecked() ? View.VISIBLE : View.GONE);
+                    setPhotoToDeleteList(position, selectToDeleteCheckBox);
+                    return true;
+                });
+    }
+
+    private void initAddMediaClickListener() {
+        RecyclerViewItemClickListener.addTo(mBinding.addPropertyMediaLayout.detailCustomViewRecyclerView, R.layout.fragment_add_property)
+                .setOnItemClickListener((recyclerView, position, v) -> {
+                    if (position == 0)
+                        Toast.makeText(requireContext(), "add", Toast.LENGTH_SHORT).show();
+                });
+    }
+
+    //___________________________________HELPERS_____________________________________________
+
```

Comment:
This method is part of the CustomView not the fragment.

Reply:
Do i have to create setter in AddPropertyCustomView? If yes, how to bind the specific instance of the custom view to set the good edit text, (AreaEditText, PriceEditText...)

Reply:
Tag the CustomView. With for example an enum of `InputType`. Use it to identify the right form field and set logic.

Reply:
ok done with static method, don't think that's what you expected but i have don't find ressources to explain on internet, so do it like i think. work fine